src/shelter/shelter.ts

```diff
@@ -189,4 +189,9 @@ export class Shelter {
 
   static isPasswordCorrect$ = (password: string) =>
     hashPassword$(password).pipe(map(passwordHash => passwordHash === Shelter._passwordHash$.getValue()));
+
+  static isPasswordCorrect = (password: string) =>
+    new Promise<boolean>(resolve => {
+      Shelter.isPasswordCorrect$(password).subscribe(resolve);
```

Comment:
this's look very weird.  
try to use Shelter.isPasswordCorrect$().toPromise() 
<img width="766" alt="image" src="https://user-images.githubusercontent.com/10281464/222417731-71bb44df-3690-4f53-94ca-38596aef71a2.png">